packages/org.json_schema.contrib/internal/TypesGenerator.pkl

```diff
@@ -60,8 +60,8 @@ function generateTypeNode(
     else
       new TypeNode.BuiltInTypeNode { type = "nothing" }
   // If we have generated a class or typealias definition already, simply use it.
-  else if (typeNames.containsKey(schema))
-    let (type = typeNames[schema as JsonSchema])
+  else if (typeNames.containsKey(schema._inline_?.getOrNull("__ref_orig__") ?? schema))
+    let (type = typeNames[(schema._inline_?.getOrNull("__ref_orig__") ?? schema) as JsonSchema])
```

Comment:
Ran into a fun case with input that looks like this:
```json
{
  "type": "object",
  "$ref": "#/$defs/users_groups.user"
}
```

Where the `schema` in the lookup `typeNames.containsKey(schema)` has `type = "object"` but the actual dereferenced schema has `type = null`. This causes a fall through to `Any` and pretty unhelpful code generation!

`ref.resolveRef` now sets `_inline_["__ref_orig__"] in its output to facilitate this lookup.